app/src/main/java/uk/gov/android/securestore/SharedPrefsStore.kt

```diff
@@ -98,67 +96,62 @@ class SharedPrefsStore(
-    ): Flow<RetrievalEvent> = callbackFlow {
-        configuration?.let { configuration ->
-            if (configuration.accessControlLevel == AccessControlLevel.OPEN) {
-                trySend(
-                    RetrievalEvent.Failed(
-                        SecureStoreErrorType.GENERAL,
-                        "Use retrieve method, access control is set to OPEN, no need for auth",
-                    ),
-                )
-            }
-            try {
-                authenticator.init(context)
-                authenticator.authenticate(
-                    configuration.accessControlLevel,
-                    authPromptConfig,
-                    AuthenticatorCallbackHandler(
-                        onSuccess = {
-                            cryptoDecryptText(key) {
-                                val event = it?.let {
-                                    RetrievalEvent.Success(it)
-                                } ?: RetrievalEvent.Failed(SecureStoreErrorType.NOT_FOUND)
-                                trySend(event)
-                            }
-                        },
-                        onError = { errorCode, errorString ->
-                            trySend(
-                                RetrievalEvent.Failed(
-                                    getErrorType(errorCode),
-                                    errorString.toString(),
-                                ),
-                            )
-                        },
-                        onFailure = {
-                            trySend(
-                                RetrievalEvent.Failed(
-                                    SecureStoreErrorType.FAILED_BIO_PROMPT,
-                                    "Bio Prompt failed",
-                                ),
-                            )
-                        },
-                    ),
-                )
-            } catch (e: GeneralSecurityException) {
-                Log.e(tag, e.message, e)
-                trySend(RetrievalEvent.Failed(SecureStoreErrorType.GENERAL))
-            } finally {
-                authenticator.close()
+    ): RetrievalEvent {
+        return configuration?.let { configuration ->
+            suspendCoroutine { continuation ->
+                if (configuration.accessControlLevel == AccessControlLevel.OPEN) {
+                    continuation.resume(
+                        RetrievalEvent.Failed(
+                            SecureStoreErrorType.GENERAL,
+                            "Use retrieve method, access control is set to OPEN, " +
+                                "no need for auth",
+                        ),
+                    )
+                }
+                try {
+                    authenticator.init(context)
+                    authenticator.authenticate(
+                        configuration.accessControlLevel,
+                        authPromptConfig,
+                        AuthenticatorCallbackHandler(
```

Comment:
I think this method could be made less complex by moving the continuation inside the AuthenticatorCallbackHandler, but It isn't a big deal. Happy to move forward without it

Reply:
Yeah, that makes sense, it would just need to reassigned to a result var then, I'll give it a go today

Reply:
I tried this, but if I use a variable to assign the result instead of .resume() there's two ways I tried and both fail, one to lateinit the var but it won't allow me as it throws a compile error and the second option to provide a default, but that means it returns the default as it doesn't wait for the AuthenticatorCallbackHandler callbacks - there might be a way I'm missing tho if you have an idea of how to go about it